subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync_iso.c

```diff
@@ -627,7 +622,20 @@ static void isr_rx(void *param)
 		e->type = EVENT_DONE_EXTRA_TYPE_SYNC_ISO_TERMINATE;
 
 		lll_isr_cleanup(param);
+
 		return;
+
+	/* Check if BIG Channel Map Update */
+	} else if (lll->chm_chan_count) {
+		uint16_t event_counter;
+
+		event_counter = lll->payload_count / lll->bn;
+		if ((event_counter - lll->ctrl_instant) <= INT16_MAX) {
+			(void)memcpy(lll->data_chan_map, lll->chm_chan_map,
+				     sizeof(lll->data_chan_map));
+			lll->data_chan_count = lll->chm_chan_count;
+			lll->chm_chan_count = 0U;
+		}
```

Comment:
I'm curious about this bit of code. 

If `(event_counter - lll->ctrl_instant) <= INT16_MAX` then it seems like we clear the value checked for a BIG channel map update (since `lll->chm_chan_count` seemingly indicates if it a channel map update). 

I assume that if `if (lll->chm_chan_count) {`, then this is a channel map update, but then why do we only conditionally clear the value? 
The condition seems to be true most of the time, given that it is only false if the difference event_counter is large (> 0x7FFF) and `lll->ctrl_instant`. 

So I am a bit puzzled on what is going on here, can you elaborate?